app/lib/utils.ts

```diff
@@ -31,3 +36,15 @@ export function getFormDataString(formData: FormData, key: string) {
     throw new Response(`expected ${key} to be a string`, { status: 400 })
   }
 }
+
+// Refreshes a given users groups and access token
+export async function refreshUser(
+  user: NonNullable<Awaited<ReturnType<typeof getUser>>>
+) {
+  const client = await getOpenIDClient()
+  const refreshedTokenSet = await client.refresh(user.refreshToken)
+  const user_new = userFromTokenSet(refreshedTokenSet)
+  await updateSession(user_new)
+  user.groups = user_new.groups
+  user.accessToken = user_new.accessToken
+}
```

Comment:
Move this to somewhere in `~/routes/__auth`? Perhaps `~/routes/__auth/user.server.ts`?

Reply:
Sounds good, will be moved in next commit